### ISSUE TITLE
Dedicated admin API for blocking a room

### ISSUE THREAD

**Issue by DMRobertson** [matrix-org/synapse#11283]
#11228 was confusing to understand, because the API call to shutdown room does several different things at once. In that change I ended up introducing a bodge, allowing admins to use that endpoint to block a room that doesn't exist.

I think there is a case for an admin API call that blocks this room and nothing else. (See also #2392).

**Comment by dklimpel** [comment]
IMO unblocking a room would be helpful, too.
Something like: 
```
POST /_synapse/admin/v1/rooms/<room_id>/block
{
    "block": true
}
```
And perhaps a list of blocked rooms:
`GET /_synapse/admin/v1/rooms/blocked`

**Comment by DMRobertson** [comment]
Closed via #11324 .
